pstoremgr/pstoremgr.go

```diff
@@ -54,38 +57,33 @@ func New(h host.Host, peerstorePath string) *Manager {
 // ImportPeer adds a new peer address to the host's peerstore, optionally
 // dialing to it. It will resolve any DNS multiaddresses before adding them.
 // The address is expected to include the /ipfs/<peerID> protocol part.
-func (pm *Manager) ImportPeer(addr ma.Multiaddr, connect bool) error {
+// Peers are added with the given ttl
+func (pm *Manager) ImportPeer(addr ma.Multiaddr, connect bool, ttl time.Duration) (peer.ID, error) {
 	if pm.host == nil {
-		return nil
+		return "", nil
 	}
 
 	logger.Debugf("adding peer address %s", addr)
-	pid, decapAddr, err := api.Libp2pMultiaddrSplit(addr)
+	pinfo, err := peerstore.InfoFromP2pAddr(addr)
 	if err != nil {
-		return err
+		return "", err
 	}
-	pm.host.Peerstore().AddAddr(pid, decapAddr, peerstore.PermanentAddrTTL)
 
-	// dns multiaddresses need to be resolved because libp2p only does that
-	// on explicit bhost.Connect().
-	if madns.Matches(addr) {
```

Comment:
By using `Connect()` below we can get rid of resolving here. `Connect` will resolve. And also use the DHT if no addresses are known for a peer. This was a bug, because `Network().DialPeer()` does not use the DHT when no addresses are known.